pkg/api/provider_client.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/models"
+	"net/http"
+	"time"
+)
+
+type ProviderClient struct{ API }
+
+func NewProviderClient(api API) *ProviderClient {
+	httpClient := http.Client{
+		Timeout: 60 * time.Second,
+	}
+
+	api.HTTPClient = httpClient
+	c := ProviderClient{API: api}
+
+	return &c
+}
+
+func (c ProviderClient) CreateAWSConnection(ctx context.Context, projectID string, conn models.AWSConnection) error {
+	b, err := json.Marshal(conn)
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("projects/%s/cloud-providers/aws/register", projectID)
+	_, err = c.doRequest(ctx, http.MethodPost, url, bytes.NewBuffer(b))
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func (c ProviderClient) GetAWSConnection(ctx context.Context, projectID string) (*models.AWSConnection, error) {
+	url := fmt.Sprintf("/projects/%s/cloud-providers/aws/register", projectID)
+	body, err := c.doRequest(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := models.AWSConnection{}
+	err = json.Unmarshal(body, &resp)
+
+	return &resp, err
+}
+
+func (c ProviderClient) RegisterAzure(ctx context.Context, projectID string, conn models.AzureConnection) error {
```

Comment:
Is there a specific reason for naming the Azure connection creation `RegisterAzure` and the AWS connection `CreateAWSConnection`? I'd suggest to unify the naming convention, as they appear to be doing the same job, right? I think either one of them is fine.

To be more precise, what I'm suggesting is, let's use either:
* `CreateAWSConnection` and `CreateAzureConnection`, or
* `RegisterAzure` and `RegisterAWS`.

Reply:
Thanks! This is my negligence.
Update to `CreateAzureConnection`